pkg/cache/clusterqueue_test.go

```diff
@@ -716,173 +716,125 @@ func TestClusterQueueUpdateWithAdmissionCheck(t *testing.T) {
-		flvResQ     resources.FlavorResourceQuantitiesFlat
-		wantDRValue int
-		wantDRName  corev1.ResourceName
+		usage               resources.FlavorResourceQuantitiesFlat
+		clusterQueue        *kueue.ClusterQueue
+		lendingClusterQueue *kueue.ClusterQueue
+		flvResQ             resources.FlavorResourceQuantitiesFlat
+		wantDRValue         int
+		wantDRName          corev1.ResourceName
 	}{
 		"no cohort": {
-			cq: ClusterQueueSnapshot{
-				FairWeight: oneQuantity,
-				Usage: resources.FlavorResourceQuantitiesFlat{
-					{Flavor: "default", Resource: corev1.ResourceCPU}: 1_000,
-					{Flavor: "default", Resource: "example.com/gpu"}:  2,
-				}.Unflatten(),
-				ResourceGroups: []ResourceGroup{
-					{
-						CoveredResources: sets.New(corev1.ResourceCPU, "example.com/gpu"),
-						Flavors: []FlavorQuotas{
-							{
-								Name: "default",
-								Resources: map[corev1.ResourceName]*ResourceQuota{
-									corev1.ResourceCPU: {
-										Nominal: 2_000,
-									},
-									"example.com/gpu": {
-										Nominal: 5,
-									},
-								},
-							},
-						},
-					},
-				},
+			usage: resources.FlavorResourceQuantitiesFlat{
+				{Flavor: "default", Resource: corev1.ResourceCPU}: 1_000,
+				{Flavor: "default", Resource: "example.com/gpu"}:  2,
 			},
+			clusterQueue: utiltesting.MakeClusterQueue("cq").
+				ResourceGroup(
+					utiltesting.MakeFlavorQuotas("default").
+						ResourceQuotaWrapper("cpu").NominalQuota("2000").Append().
+						ResourceQuotaWrapper("example.com/gpu").NominalQuota("5").Append().
+						FlavorQuotas,
+				).Obj(),
 		},
 		"usage below nominal": {
-			cq: ClusterQueueSnapshot{
-				FairWeight: oneQuantity,
-				Usage: resources.FlavorResourceQuantitiesFlat{
-					{Flavor: "default", Resource: corev1.ResourceCPU}: 1_000,
-					{Flavor: "default", Resource: "example.com/gpu"}:  2,
-				}.Unflatten(),
-				ResourceGroups: []ResourceGroup{
-					{
-						CoveredResources: sets.New(corev1.ResourceCPU, "example.com/gpu"),
-						Flavors: []FlavorQuotas{
-							{
-								Name: "default",
-								Resources: map[corev1.ResourceName]*ResourceQuota{
-									corev1.ResourceCPU: {
-										Nominal: 2_000,
-									},
-									"example.com/gpu": {
-										Nominal: 5,
-									},
-								},
-							},
-						},
-					},
-				},
-				Cohort: &CohortSnapshot{
-					Lendable: map[corev1.ResourceName]int64{
-						corev1.ResourceCPU: 10_000,
-						"example.com/gpu":  10,
-					},
-				},
+			usage: resources.FlavorResourceQuantitiesFlat{
+				{Flavor: "default", Resource: corev1.ResourceCPU}: 1_000,
+				{Flavor: "default", Resource: "example.com/gpu"}:  2,
 			},
+			clusterQueue: utiltesting.MakeClusterQueue("cq").
+				Cohort("test-cohort").
+				FairWeight(oneQuantity).
+				ResourceGroup(
+					utiltesting.MakeFlavorQuotas("default").
+						ResourceQuotaWrapper("cpu").NominalQuota("2").LendingLimit("0").Append().
+						ResourceQuotaWrapper("example.com/gpu").NominalQuota("5").LendingLimit("0").Append().
```

Comment:
Why setting `lendingLimit` for the resources - I don't see it in the red (removed) code? Wouldn't it be beneficial for the main clusterQueue to have the same config as before?

Reply:
Because this capacity would then be part of the Cohort's lendable. We could remove it, but then we would have to change the nomialQuota of the lendingClusterQueues

Reply:
I see, to make the lendingClusterQueue truly complementary. TBH my preference is to do so, if not too much work, to make sure there are no subtle changes in coverage.

Reply:
Yup, very easy change to make.

Done.